app/Services/CreditCheckService.php

```diff
@@ -49,43 +50,49 @@ public static function accountNumberVerification(
         string $bank_name,
         string $reference = null
     ) {
-        try {
-            $isValid = false;
-            $orderQuery = NewOrder::query();
-            $order = is_string($order_id) ? $orderQuery->where('order_number', $order_id)->first() : $orderQuery->where('id', $order_id)->first();
-            $latestCreditReport = Recommendation::query()->where('customer_id', $customer_id)->where('type', 'credit_report')->latest('created_at')->first();
-            if ($latestCreditReport) {
-                $data = json_decode($latestCreditReport->input_data);
-                if (property_exists($data, 'accountName') && property_exists($data, 'bankName')) {
-                    $isValid = $data->accountName == $account_name  && $data->bankName ==  $bank_name;
-                }
-                if (!$isValid) {
-                    // dd($data);
-                    //keep a record 
-                   $missMatchedPayment =  MissMatchedPayments::create([
-                        'reference' => $reference,
-                        'customer_id' => $customer_id,
-                        'order_id' => $order->id,
+        if (env('USE_MISS_MATCHED_ACCOUNT')) {
```

Comment:
Can we add it where we are calling the function since it covers the whole block?

Reply:
We are calling the function in 3 different places; I feel having the switch in one place is okay.

OR why apply individually?

Reply:
Let's do the check at the top of the function then and return if not set, so it's cleaner.